File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListBox/Functionality/DragDropPage.vb
Namespace CompanionKit.Controls.ListBox.Functionality

	Public Class DragDropPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()



		End Sub

        ''' <summary>
        ''' Handles the DragDrop event of the mobjListBox2 control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="global.Gizmox.WebGUI.Forms.DragEventArgs"/> instance containing the event data.</param>
        Private Sub mobjListBox2_DragDrop(ByVal sender As System.Object, ByVal e As Global.Gizmox.WebGUI.Forms.DragEventArgs) Handles mobjListBox2.DragDrop
            If TypeOf e Is DragDropEventArgs Then
                Dim dragDropEventArgs As DragDropEventArgs = TryCast(e, DragDropEventArgs)
                If TypeOf dragDropEventArgs.Source Is Global.Gizmox.WebGUI.Forms.ListBox Then
                    Dim addedItem As String = TryCast(DirectCast(dragDropEventArgs.Source, Global.Gizmox.WebGUI.Forms.ListBox).SelectedItem, String)
                    If (Not addedItem Is Nothing) Then
                        Me.mobjListBox2.Items.Add(addedItem)
                    End If
                End If
            End If
        End Sub

        ''' <summary>
        ''' Handles the CheckedChanged event of the mobjAllowDrop control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjAllowDrop_CheckedChanged(sender As System.Object, e As System.EventArgs) Handles mobjAllowDrop.CheckedChanged
            mobjListBox2.AllowDrop = mobjAllowDrop.Checked
        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/ListBox/Functionality/DragDropPage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.ListBox.Functionality
{
    public partial class DragDropPage : UserControl
    {
        public DragDropPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the DragDrop event of the mobjListBox2 control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="DragEventArgs"/> instance containing the event data.</param>
        private void mobjListBox2_DragDrop(object sender, DragEventArgs e)
        {
            if (e is DragDropEventArgs)
            {
                DragDropEventArgs dragDropEventArgs = e as DragDropEventArgs;
                if (dragDropEventArgs.Source is Gizmox.WebGUI.Forms.ListBox)
                {
                    //Get text of drag item of left ListBox.
                    string addedItem = ((Gizmox.WebGUI.Forms.ListBox)dragDropEventArgs.Source).SelectedItem as string;
                    if (addedItem != null)
                    {
                        //Add draged item to right ListBox
                        this.mobjListBox2.Items.Add(addedItem);
                    }
                }
            }
        }

        /// <summary>
        /// Handles the CheckedChanged event of the mobjAllowDrop control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjAllowDrop_CheckedChanged(object sender, EventArgs e)
        {
            mobjListBox2.AllowDrop = mobjAllowDrop.Checked;
        }

    }
}